src/allmydata/test/test_node.py

```diff
@@ -134,26 +135,44 @@ def test_private_config(self):
         f.write("secret")
         f.close()
 
-        n = TestNode(basedir)
-        self.failUnlessEqual(n.get_private_config("already"), "secret")
-        self.failUnlessEqual(n.get_private_config("not", "default"), "default")
-        self.failUnlessRaises(MissingConfigEntry, n.get_private_config, "not")
-        value = n.get_or_create_private_config("new", "start")
+        config = config_from_string("", "", basedir)
+
+        self.failUnlessEqual(config.get_private_config("already"), "secret")
+        self.failUnlessEqual(config.get_private_config("not", "default"), "default")
+        self.failUnlessRaises(MissingConfigEntry, config.get_private_config, "not")
+        value = config.get_or_create_private_config("new", "start")
         self.failUnlessEqual(value, "start")
-        self.failUnlessEqual(n.get_private_config("new"), "start")
+        self.failUnlessEqual(config.get_private_config("new"), "start")
         counter = []
         def make_newer():
             counter.append("called")
             return "newer"
-        value = n.get_or_create_private_config("newer", make_newer)
+        value = config.get_or_create_private_config("newer", make_newer)
         self.failUnlessEqual(len(counter), 1)
         self.failUnlessEqual(value, "newer")
-        self.failUnlessEqual(n.get_private_config("newer"), "newer")
+        self.failUnlessEqual(config.get_private_config("newer"), "newer")
 
-        value = n.get_or_create_private_config("newer", make_newer)
+        value = config.get_or_create_private_config("newer", make_newer)
         self.failUnlessEqual(len(counter), 1) # don't call unless necessary
         self.failUnlessEqual(value, "newer")
 
+    def test_write_config_unwritable_file(self):
+        basedir = "test_node/configdir"
+        fileutil.make_dirs(basedir)
+        config = config_from_string("", "", basedir)
+        with open(os.path.join(basedir, "bad"), "w") as f:
+            f.write("bad")
+        os.chmod(os.path.join(basedir, "bad"), 0o000)
+
+        config.write_config_file("bad", "some value")
+
+        errs = self.flushLoggedErrors()
+        self.assertEqual(1, len(errs))
+        self.assertIn(
+            "IOError",
+            str(errs[0])
+        )
```

Comment:
If you want, I'm pretty sure you could `flushLoggedErrors(IOError)` - then the length assertion is probably enough.